src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_WORD = "supply";
+    private static final String BUY_WORD = "buy";
+    private static final String SEPARATION_SIGN = ",";
+    private static final String RESULT_WORD = "result";
+    private static final int TYPE_OF_OPERATION_INDEX = 0;
+    private static final int AMOUNT_OF_OPERATION_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readFromFile(fromFileName);
+        String finalReport = createReport(data);
+        writeToFile(finalReport, toFileName);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read a file with name " + fromFileName, e);
+        }
+        return stringBuilder.toString().split(System.lineSeparator());
+    }
+
+    private String createReport(String[] data) {
+        int buyAmount = 0;
+        int supplyAmount = 0;
+        int result;
```

Comment:
there is no sense of declaring this variable beforehand - you can declare it directly on line 51